transaction-view/src/transaction_view.rs

```diff
@@ -7,26 +7,39 @@ use {
     solana_sdk::{hash::Hash, pubkey::Pubkey, signature::Signature},
 };
 
+// alias for convenience
+pub type UnsanitizedTransactionView<D> = TransactionView<false, D>;
+pub type SanitizedTransactionView<D> = TransactionView<true, D>;
+
 /// A view into a serialized transaction.
 ///
 /// This struct provides access to the transaction data without
 /// deserializing it. This is done by parsing and caching metadata
 /// about the layout of the serialized transaction.
 /// The owned `data` is abstracted through the `TransactionData` trait,
 /// so that different containers for the serialized transaction can be used.
-pub struct TransactionView<D: TransactionData> {
-    data: D,
-    meta: TransactionMeta,
+pub struct TransactionView<const SANITIZED: bool, D: TransactionData> {
+    pub(crate) data: D,
+    pub(crate) meta: TransactionMeta,
```

Comment:
I would like these to stay private since they are quite sensitive. I could be convinced that `pub(crate)` is safe and fine but it doesn't seem necessary.